src/main/java/com/foundation/search/view/ParametersPanel.java

```diff
@@ -156,7 +340,84 @@ public JTextField getNameInput(){
      *
      * @return extensionInput
      */
-    public JTextField getExtensionInput(){
-        return extensionInput;
+    public String getExtensionInput() {
+        return extensionInput.getSelectedItem().toString();
+    }
+
+    /**
+     * This method returns an array of String with information to search by size
+     *
+     * @return sizeData
+     */
+    public String[] getSizeInput(){
+        sizeData = new String[2];
+        sizeData[0] = sizeComparison.getSelectedItem().toString();
+        sizeData[1] = sizeInput.getText();
+        sizeData[2] = sizeUnit.getSelectedItem().toString();
+        return sizeData;
     }
+
+    /**
+     * This method listens the actions in the checkboxes. overwrites the method
+     * existent in the ITemListener interface
+     *
+     * @param ItemEvent it is the event
+     */
+    @Override
+    public void itemStateChanged(ItemEvent e) {
+        Object source = e.getItemSelectable();
+
+        if (source == hidden) {
+            System.out.println("hidden is true");
+            hiddenCH = true;
+        } else if (source == read_only) {
+            System.out.println("read only is true");
+            read_onlyCH = true;
+        } else if (source == only_files) {
+            System.out.println("only files is true");
+            only_filesCH = true;
+        }
+
+        if (e.getStateChange() == ItemEvent.DESELECTED){
+            System.out.println("itemevent is deselected");
+            if (source == hidden) {
+                System.out.println("hidden is false again");
+                hiddenCH = false;
+            } else if (source == read_only) {
+                System.out.println("read only is false again");
+                read_onlyCH = false;
+            } else if (source == only_files) {
+                System.out.println("only files is false again");
+                only_filesCH = false;
+            }
+        }
+    }
+
+    /**
+     * This method returns the value of Hidden checkbox
+     *
+     * return boolean hiddenCH
+     */
+    public boolean getHidden(){
+        return hiddenCH;
+    }
+
+    /**
+     * This method returns the value of Hidden checkbox
```

Comment:
1. You forgot to update the description, instead of "hidden" should be Read Only
2. You are missing the @ at the beginning of the return word also --> should be: @return boolean readOnlyCh
Same observation for line 415 and 417

Reply:
updated both!